### ISSUE TITLE
gt and gT not working in VS2019

### ISSUE THREAD

**Issue by marza91** [jaredpar/VsVim#2481]
Like the title says, I can't change tabs anymore with gt or gT. No error message as far as I can see, and nothing happens. Tried in multiple different projects and to remove my _vimrc just to be sure.

Paste of my version info:

Microsoft Visual Studio Enterprise 2019 
Version 16.0.0
VisualStudio.16.Release/16.0.0+28729.10
Microsoft .NET Framework
Version 4.7.03056

Installed Version: Enterprise

ADL Tools Service Provider   1.0
This package contains services used by Data Lake tools

Application Insights Tools for Visual Studio Package   9.0.20307.1
Application Insights Tools for Visual Studio

ASP.NET and Web Tools 2019   16.0.12311.10635
ASP.NET and Web Tools 2019

ASP.NET Web Frameworks and Tools 2019   16.0.12311.10635
For additional information, visit https://www.asp.net/

Azure App Service Tools v3.0.0   16.0.12311.10635
Azure App Service Tools v3.0.0

Azure Data Lake Node   1.0
This package contains the Data Lake integration nodes for Server Explorer.

Azure Data Lake Tools for Visual Studio   2.3.8000.0
Microsoft Azure Data Lake Tools for Visual Studio

Azure Functions and Web Jobs Tools   16.0.12311.10635
Azure Functions and Web Jobs Tools

Azure Stream Analytics Tools for Visual Studio   2.3.8000.0
Microsoft Azure Stream Analytics Tools for Visual Studio

C# Tools   3.0.0-beta4-19170-01+1deafee3682da88bf07d1c18521a99f47446cee8
C# components used in the IDE. Depending on your project type and settings, a different version of the compiler may be used.

Common Azure Tools   1.10
Provides common services for use by Azure Mobile Services and Microsoft Azure Tools.

Extensibility Message Bus   1.1.77 (master@24013d5)
Provides common messaging-based MEF services for loosely coupled Visual Studio extension components communication and integration.

Fabric.DiagnosticEvents   1.0
Fabric Diagnostic Events

Microsoft Azure HDInsight Azure Node   2.3.8000.0
HDInsight Node under Azure Node

Microsoft Azure Hive Query Language Service   2.3.8000.0
Language service for Hive query

Microsoft Azure Service Fabric Tools for Visual Studio   2.5
Microsoft Azure Service Fabric Tools for Visual Studio

Microsoft Azure Stream Analytics Language Service   2.3.8000.0
Language service for Azure Stream Analytics

Microsoft Azure Stream Analytics Node   1.0
Azure Stream Analytics Node under Azure Node

Microsoft Azure Tools   2.9
Microsoft Azure Tools for Microsoft Visual Studio 0x10 - v2.9.0.0

Microsoft Continuous Delivery Tools for Visual Studio   0.4
Simplifying the configuration of Azure DevOps pipelines from within the Visual Studio IDE.

Microsoft JVM Debugger   1.0
Provides support for connecting the Visual Studio debugger to JDWP compatible Java Virtual Machines

Microsoft Library Manager   1.0
Install client-side libraries easily to any web project

Microsoft MI-Based Debugger   1.0
Provides support for connecting Visual Studio to MI compatible debuggers

Microsoft Visual Studio Tools for Containers   1.1
Develop, run, validate your ASP.NET Core applications in the target environment. F5 your application directly into a container with debugging, or CTRL + F5 to edit & refresh your app without having to rebuild the container.

Mono Debugging for Visual Studio   16.0.300 (573eda3)
Support for debugging Mono processes with Visual Studio.

NuGet Package Manager   5.0.0
NuGet Package Manager in Visual Studio. For more information about NuGet, visit http://docs.nuget.org/.

ProjectServicesPackage Extension   1.0
ProjectServicesPackage Visual Studio Extension Detailed Info

ResourcePackage Extension   1.0
ResourcePackage Visual Studio Extension Detailed Info

ResourcePackage Extension   1.0
ResourcePackage Visual Studio Extension Detailed Info

Snapshot Debugging Extension   1.0
Snapshot Debugging Visual Studio Extension Detailed Info

SQL Server Data Tools   16.0.61903.25110
Microsoft SQL Server Data Tools

ToolWindowHostedEditor   1.0
Hosting json editor into a tool window

TypeScript Tools   16.0.10306.2001
TypeScript Tools for Microsoft Visual Studio

Visual Basic Tools   3.0.0-beta4-19170-01+1deafee3682da88bf07d1c18521a99f47446cee8
Visual Basic components used in the IDE. Depending on your project type and settings, a different version of the compiler may be used.

Visual F# Tools 10.4 for F# 4.6   16.0.0.0.  Commit Hash: 809f41207b74a1356208f11b8fa3a15233734050.
Microsoft Visual F# Tools 10.4 for F# 4.6

Visual Studio Code Debug Adapter Host Package   1.0
Interop layer for hosting Visual Studio Code debug adapters in Visual Studio

Visual Studio Tools for Containers   1.0
Visual Studio Tools for Containers

Visual Studio Tools for Kubernetes   1.0
Visual Studio Tools for Kubernetes

VisualStudio.Mac   1.0
Mac Extension for Visual Studio

VsVim   2.6.0.0
VsVim is a Vim emulator for Visual Studio

Web Compiler   1.12.394
Compiler for LESS, Sass and CoffeeScript files

Xamarin   16.0.0.513 (d16-0@aebc6bdb5)
Visual Studio extension to enable development for Xamarin.iOS and Xamarin.Android.

Xamarin Designer   4.17.4.413 (remotes/origin/d16-0@db42fb8b0)
Visual Studio extension to enable Xamarin Designer tools in Visual Studio.

Xamarin Templates   16.1.43 (fe3ade5)
Templates for building iOS, Android, and Windows apps with Xamarin and Xamarin.Forms.

Xamarin.Android SDK   9.2.0.5 (HEAD/292d27fa)
Xamarin.Android Reference Assemblies and MSBuild support.
    Mono: mono/mono/2018-08-rc@5ad371dab1b
    Java.Interop: xamarin/java.interop/d16-0@c987483
    LibZipSharp: grendello/LibZipSharp/master@44de300
    LibZip: nih-at/libzip/rel-1-5-1@b95cf3f
    MXE: xamarin/mxe/xamarin@b9cbb535
    ProGuard: xamarin/proguard/master@905836d
    SQLite: xamarin/sqlite/3.26.0@325e91a
    Xamarin.Android Tools: xamarin/xamarin-android-tools/d16-0@0a7edd6

Xamarin.iOS and Xamarin.Mac SDK   12.6.0.23 (ad57f87)
Xamarin.iOS and Xamarin.Mac Reference Assemblies and MSBuild support.

**Comment by marza91** [comment]
Found the setting for logging to the output window, and recorded what happens when I press gt and then gT:

```
VsVim VimKeyProcessor::KeyDown G None None
VsVim VimKeyProcessor::KeyDown Handled = False
VsVim VimKeyProcessor::TextInput Text=g ControlText= SystemText=
VsVim VimKeyProcessor::TextInput Handled=True
VsVim VimBufferCoordinator::PreviewKeyUp G
VsVim VimKeyProcessor::KeyUp G None
VsVim VimKeyProcessor::KeyUp Handled = False
VsVim VimBufferCoordinator::KeyUp G
VsVim VimKeyProcessor::KeyDown T None None
VsVim VimKeyProcessor::KeyDown Handled = False
VsVim VimKeyProcessor::TextInput Text=t ControlText= SystemText=
VsVim VimKeyProcessor::TextInput Handled=True
VsVim VimBufferCoordinator::PreviewKeyUp T
VsVim VimKeyProcessor::KeyUp T None
VsVim VimKeyProcessor::KeyUp Handled = False
VsVim VimBufferCoordinator::KeyUp T
VsVim VimKeyProcessor::KeyDown G None None
VsVim VimKeyProcessor::KeyDown Handled = False
VsVim VimKeyProcessor::TextInput Text=g ControlText= SystemText=
VsVim VimKeyProcessor::TextInput Handled=True
VsVim VimBufferCoordinator::PreviewKeyUp G
VsVim VimKeyProcessor::KeyUp G None
VsVim VimKeyProcessor::KeyUp Handled = False
VsVim VimBufferCoordinator::KeyUp G
VsVim VimKeyProcessor::KeyDown RightShift None Shift
VsVim VimKeyProcessor::KeyDown Handled = False
VsVim VimKeyProcessor::KeyDown T None Shift
VsVim VimKeyProcessor::KeyDown Handled = False
VsVim VimKeyProcessor::TextInput Text=T ControlText= SystemText=
VsVim VimKeyProcessor::TextInput Handled=True
VsVim VimBufferCoordinator::PreviewKeyUp T
VsVim VimKeyProcessor::KeyUp T Shift
VsVim VimKeyProcessor::KeyUp Handled = False
VsVim VimBufferCoordinator::KeyUp T
VsVim VimBufferCoordinator::PreviewKeyUp RightShift
VsVim VimKeyProcessor::KeyUp RightShift None
VsVim VimKeyProcessor::KeyUp Handled = False
VsVim VimBufferCoordinator::KeyUp RightShift
```

I can also see a lot of lines like this popping up over and over, not sure if it's related to anything:

```
VsVim VsCommandTarget::QueryStatus Comment Key.None: 1496a755-94de-11d0-8c3f-00c04fc2aae2 136
VsVim VsCommandTarget::QueryStatus 
VsVim VsCommandTarget::QueryStatus Uncomment Key.None: 1496a755-94de-11d0-8c3f-00c04fc2aae2 137
VsVim VsCommandTarget::QueryStatus 
VsVim VsCommandTarget::QueryStatus Comment Key.None: 1496a755-94de-11d0-8c3f-00c04fc2aae2 136
VsVim VsCommandTarget::QueryStatus 
VsVim VsCommandTarget::QueryStatus Uncomment Key.None: 1496a755-94de-11d0-8c3f-00c04fc2aae2 137
VsVim VsCommandTarget::QueryStatus 
VsVim VsCommandTarget::QueryStatus Comment Key.None: 1496a755-94de-11d0-8c3f-00c04fc2aae2 136
VsVim VsCommandTarget::QueryStatus 
VsVim VsCommandTarget::QueryStatus Uncomment Key.None: 1496a755-94de-11d0-8c3f-00c04fc2aae2 137
VsVim VsCommandTarget::QueryStatus 
VsVim VsCommandTarget::QueryStatus Comment Key.None: 1496a755-94de-11d0-8c3f-00c04fc2aae2 136
VsVim VsCommandTarget::QueryStatus 
VsVim VsCommandTarget::QueryStatus Uncomment Key.None: 1496a755-94de-11d0-8c3f-00c04fc2aae2 137
VsVim VsCommandTarget::QueryStatus 
VsVim VsCommandTarget::QueryStatus Comment Key.None: 1496a755-94de-11d0-8c3f-00c04fc2aae2 136
VsVim VsCommandTarget::QueryStatus 
VsVim VsCommandTarget::QueryStatus Uncomment Key.None: 1496a755-94de-11d0-8c3f-00c04fc2aae2 137
VsVim VsCommandTarget::QueryStatus 
VsVim VsCommandTarget::QueryStatus Comment Key.None: 1496a755-94de-11d0-8c3f-00c04fc2aae2 136
VsVim VsCommandTarget::QueryStatus 
VsVim VsCommandTarget::QueryStatus Uncomment Key.None: 1496a755-94de-11d0-8c3f-00c04fc2aae2 137
VsVim VsCommandTarget::QueryStatus 
VsVim VsCommandTarget::QueryStatus Comment Key.None: 1496a755-94de-11d0-8c3f-00c04fc2aae2 136
VsVim VsCommandTarget::QueryStatus 
VsVim VsCommandTarget::QueryStatus Uncomment Key.None: 1496a755-94de-11d0-8c3f-00c04fc2aae2 137
VsVim VsCommandTarget::QueryStatus 
```

**Comment by leechiryo** [comment]
Just to report that I met the same problem.
Hope it will be fixed soon.

**Comment by AbdulMuqsit** [comment]
Facing the same issue.

**Comment by TKumadaki** [comment]
Hi @marza91
Your problem may be solved by installing the latest VsVim.
http://vsixgallery.com/extension/VsVim.Microsoft.e214908b-0458-4ae2-a583-4310f29687c3/

**Comment by jaredpar** [comment]
Hmm. I wonder if the APIs I rely on for changing the windows changed in a more recent build. I'm not quite on the latest VS2019 at the moment. Will need to update before I can dig into this.

**Comment by marza91** [comment]
Hi again, I tested with the linked version and it seems to be working again!
Now running version 2.7.1324

**Comment by ricksladkey** [comment]
Ha, ha, finding out about `gt` and `gT` is reason enough to read all the issues.

**Comment by ricksladkey** [comment]
As @TKumadaki pointed out, this is caused by using the published 2.6.0 version of VsVim which claims that it supports VS2019, but which does not in fact support any Visual Studio APIs that are accessed through VsVim's version-specific assemblies. This deficiency was fixed in PR #2447 and so will be included in 2.7.0 when it is released.
